File: android/app/src/main/java/com/jamesbrown/aflmobile/data/repository/AppContainer.kt
package com.jamesbrown.aflmobile.data.repository

import android.content.Context
import com.jamesbrown.aflmobile.data.network.BackendApiClient
import com.jamesbrown.aflmobile.data.settings.AppSettingsStore
import kotlinx.coroutines.CoroutineScope
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.SupervisorJob


class AppContainer(context: Context) {
    private val applicationScope = CoroutineScope(SupervisorJob() + Dispatchers.IO)
    val settingsStore = AppSettingsStore(context)
    private val apiClient = BackendApiClient(settingsStore)
    val repository = AflRepository(apiClient, settingsStore)
    val sgmDraftStore = SgmDraftStore(
        persistence = DataStoreSgmDraftPersistence(
            saveJson = settingsStore::saveSgmDraftJson,
            loadJson = settingsStore::sgmDraftJson,
        ),
        scope = applicationScope,
    )
}

File: macos/Sources/AFLEdgeMac/App/AppContainer.swift
import Foundation
import Observation

@MainActor
@Observable
final class AppContainer {
    let settingsStore: AppSettingsStore
    let apiClient: AFLAPIClient
    let sgmDraftStore: SgmDraftStore
    let cgmDraftStore: CgmDraftStore

    init(settingsStore: AppSettingsStore = AppSettingsStore()) {
        self.settingsStore = settingsStore
        self.apiClient = AFLAPIClient(settingsProvider: { settingsStore.settings })
        self.sgmDraftStore = SgmDraftStore()
        self.cgmDraftStore = CgmDraftStore()
    }
}
